tfrobot/pkg/deployer/deployer.go

```diff
@@ -250,12 +258,20 @@ func buildDeployments(vms []Vms, nodesIDs []int, sshKeys map[string]string) grou
 			volumes, volumeMounts := parseVolumes(vmName, vmGroup.Volumes)
 
 			network := buildNetworkDeployment(vmGroup, nodeID, vmName, solutionType)
-			vm := buildVMDeployment(vmGroup, nodeID, vmName, network.Name, sshKeys[vmGroup.SSHKey], append(diskMounts, volumeMounts...))
-
-			deployment := workloads.NewDeployment(vm.Name, nodeID, solutionType, nil, network.Name, disks, nil, []workloads.VM{vm}, nil, nil, volumes)
+			deployment := workloads.NewDeployment("", nodeID, solutionType, nil, network.GetName(), disks, nil, nil, nil, nil, volumes)
+
+			if !vmGroup.WireGuard && !vmGroup.PublicIP4 && !vmGroup.PublicIP6 && !vmGroup.Ygg {
+				vm := buildVMLightDeployment(vmGroup, nodeID, vmName, network.GetName(), sshKeys[vmGroup.SSHKey], append(diskMounts, volumeMounts...))
+				deployment.VmsLight = append(deployment.VmsLight, vm)
+				deployment.Name = vm.Name
+			} else {
+				vm := buildVMDeployment(vmGroup, nodeID, vmName, network.GetName(), sshKeys[vmGroup.SSHKey], append(diskMounts, volumeMounts...))
+				deployment.Vms = append(deployment.Vms, vm)
+				deployment.Name = vm.Name
+			}
```

Comment:
should we have a function `buildDeployment` similar to `buildNetworkDeployment` and move all the logic related to building the vms to